src/bootstrap/lib.rs

```diff
@@ -463,6 +463,14 @@ impl Build {
             bootstrap_out
         };
 
+        if !rust_info.is_git() {
+            config.description = if let Some(description) = config.description {
+                Some(format!("{description}, built from a source tarball"))
```

Comment:
hmm, I'm not sure this is the right behavior here. My understanding of Mark's suggestion in https://github.com/rust-lang/rust/pull/100557#issuecomment-1214874860 was:
1. If we're building from git sources, leave `description` blank by default
2. If we're building from a tarball, make `description` default to `built from a source tarball`, but allow it to be overriden
3. Add a comment to `config.toml.example` recommending that you turn off `git-commit-hash` handling when applying patches, and saying you can explicitly set `description` to an empty string so you can opt-in to using pre-compiled libstd binaries from static.rlo if you want to.

If we don't allow overriding the description in 2, then we don't actually solve https://github.com/rust-lang/rust/issues/33286 because the version will still differ from the official binaries.
```suggestion
                Some(format!("{description}"))
```

Reply:
> Add a comment to config.toml.example recommending that you turn off git-commit-hash handling when applying patches, and saying you can explicitly set description to an empty string so you can opt-in to using pre-compiled libstd binaries from static.rlo if you want to.

I've made a comment doing the second locally, but how would you turn of `git-commit-info` handling? I don't think this has an option for that right now.

Reply:
That's what I meant by this second step:
> If we're building from a tarball, make description default to built from a source tarball, but allow it to be overriden

We need to make it possible for people to turn off the handling; you can do that by avoiding adding the "built from source tarball" text when `description` is non-empty.